lib/awful/rules.lua

```diff
@@ -195,23 +212,29 @@ end
 -- @tab props Properties to apply.
 -- @tab[opt] callbacks Callbacks to apply.
 function rules.execute(c, props, callbacks)
+    local switchtotag = props.switchtotag
     for property, value in pairs(props) do
         if property ~= "focus" and type(value) == "function" then
             value = value(c)
         end
         if property == "floating" then
             aclient.floating.set(c, value)
+        elseif property == "screen" then
+            -- Support specifying screens by name ("VGA1")
+            c.screen = screen[value]
         elseif property == "tag" then
-            c.screen = atag.getscreen(value)
-            c:tags({ value })
-        elseif property == "switchtotag" and value and props.tag then
-            atag.viewonly(props.tag)
+            local t = value
+            if type(t) == "string" then
+                t = atag.find_by_name(props.screen, t)
+            end
+            c.screen = atag.getscreen(t)
+            c:tags({ t })
         elseif property == "height" or property == "width" or
                 property == "x" or property == "y" then
             local geo = c:geometry();
             geo[property] = value
             c:geometry(geo);
-        elseif property == "focus" then
+        elseif property == "focus" or property == "switchtotag" then
             -- This will be handled below
             (function() end)() -- I haven't found a nice way to silence luacheck here
```

Comment:
This is to get rid of the "I haven't found a nice way to silence luacheck here", right?
Yeah, that could work. I'll (hopefully remember to) do a separate PR for that. I guess I would do:
```lua
local ignore = { focus = true, switchtotag = true }
....
elseif not ignore[property] then
    if type(c[property]) == "function" then
        c[property](c, value)
    else
        c[property] = value
    end
end
```